runner/runner.go

```diff
@@ -0,0 +1,252 @@
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// NewRunner returns an initialized instance of *Runner.
+func NewRunner(role string) *Runner {
+	return &Runner{
+		role: role,
+	}
+}
+
+// Response provides a command Response.
+type Response struct {
+	name        string
+	duration    string
+	list        bool
+
+	output string
+	code   int
+	description string
+	fullCmd []string
+	timeout string
+}
+
+type response struct {
+	Output      string `json:"output"`
+	Code        int `json:"code"`
+}
+
+type responseList struct {
+	Description string `json:"description"`
+	FullCmd     []string `json:"full_cmd"`
+	Timeout     string `json:"timeout"`
+}
+
+// MarshalJSON is a custom json marshaller implementation used to return output based on user request.
+// responseList is returned if a user requested to list the checks. response is used to return back the result
+// with populated Output and executable return code.
+func (r Response) MarshalJSON() ([]byte, error) {
+	if r.list {
+		return json.Marshal(&responseList{
+			Description: r.description,
+			FullCmd: r.fullCmd,
+			Timeout: r.timeout,
+		})
+	}
+
+	return json.Marshal(&response{
+		Output: r.output,
+		Code: r.code,
+	})
+}
+
+// NewCombinedResponse initiates a new instance of CombinedResponse.
+func NewCombinedResponse(list bool) *CombinedResponse {
+	return &CombinedResponse{
+		checks: make(map[string]*Response),
+		errs:   make(map[string]*Response),
+		list: list,
+	}
+}
+
+// CombinedResponse represents a parent structure for multiple checks.
+type CombinedResponse struct {
+	status int
+	list   bool
+	checks map[string]*Response
+	errs   map[string]*Response
+}
+
+// Status returns checks combined status.
+func (cr CombinedResponse) Status() int {
+	return cr.status
+}
+
+// MarshalJSON is a custom json marshaller implementation used to return the appropriate response based
+// on user input. combinedResponseError is used to return back error message if runner was unable to execute a check.
+// CombinedResponse.Checks is used to return back a list of checks without executing them, combinedResponseSuccess is
+// used to return user the actual checks output.
+func (cr CombinedResponse) MarshalJSON() ([]byte, error) {
+	if len(cr.errs) > 0 {
+		var errs []string
+		for e, r := range cr.errs {
+			errs = append(errs, fmt.Sprintf("%s: %s", r.name, e))
+		}
+
+		return json.Marshal(combinedResponseError{
+			Error: "One or more requested checks failed to execute.",
+			Checks: errs,
+		})
+	}
+
+	if cr.list {
+		return json.Marshal(cr.checks)
+	}
+
+	return json.Marshal(combinedResponseSuccess{
+		Status: cr.status,
+		Checks: cr.checks,
+	})
+}
+
+type combinedResponseSuccess struct {
+	Status int `json:"status"`
+	Checks map[string]*Response `json:"checks"`
+}
+
+type combinedResponseError struct {
+	Error string `json:"error"`
+	Checks []string `json:"checks"`
+}
+
+// Runner is a main instance of DC/OS runner runner.
+type Runner struct {
```

Comment:
In the future it would be nice if we listed our structs at the top. I think @klueska has said something similar, where the structure of the file should generally be

constants
vars
types
functions

etc